src/index.ts

```diff
@@ -123,9 +123,10 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
         const templateContext = {
             define: htmlFileConfiguration.define,
         }
-
-        const compiledTemplateFn = lodashTemplate(htmlTemplate)
-        return compiledTemplateFn(templateContext)
+        const compiledTemplateFn = htmlTemplate.toString().endsWith('.html') 
+            ? lodashTemplate(fs.readFileSync(htmlTemplate).toString()) 
```

Comment:
I think we can safely convert renderTemplate to be async, and use `await fs.readFile` from the promises package instead of renaming all the references of fs to promises.